zygrader/zybooks.py

```diff
@@ -31,8 +32,31 @@ class Zybooks:
     def __init__(self):
         Zybooks.session = requests.session()
 
+    def __load_session(self):
+        return preferences.get("token")
+
+    def __save_session(self, token: str):
+        preferences.set("token", token)
+
+    def __check_auth(self):
+        """Ensure the auth token is valid through a "useless" request"""
+        check_url = f'https://zyserver.zybooks.com/v1/zybooks?zybooks=["{SharedData.CLASS_CODE}"]'
+        check_url += f"&auth_token={Zybooks.token}"
+        r = Zybooks.session.get(check_url)
```

Comment:
I feel like `requests` probably has some system for passing url params in the query string. that would probably be cleaner than the url building here. if it doesn't exist then this is fine